File: projecteuler/python/problem_7.py
'''
Created on 03/09/2011

@author: Breno Martinusso

By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see that the 6th prime is 13.

What is the 10 001st prime number?

Awnser: 104743
'''

def is_prime_number(number):
    for n in xrange(2, number/2 +1):
        if not number % n:
            return False
    return True

n, i = 1, 0
while i < 10001:
    n += 1
    if is_prime_number(n):
        i += 1
print n

File: projecteuler/go/problem7.go
package projecteuler

// Problem 7
// 10001st prime
// By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see that the 6th prime is 13.
//
// What is the 10 001st prime number?

const (
	answerProblem7 = 104743
)

func problem7() int64 {
	var (
		count     int
		lastPrime int64
	)

	for i := 0; count < 10001; i++ {
		if isPrime(i) {
			lastPrime = int64(i)
			count++
		}
	}
	return lastPrime
}

func isPrime(number int) bool {
	if number < 2 {
		return false
	}
	if number == 2 {
		return true
	}

	for i := 2; i <= (number / 2); i++ {
		if number%i == 0 {
			return false
		}
	}
	return true
}
